src/NUnitFramework/framework/Internal/ExceptionHelper.cs

```diff
@@ -33,6 +34,41 @@ namespace NUnit.Framework.Internal
     /// </summary>
     public class ExceptionHelper
     {
+#if !NET_4_5 && !PORTABLE
+        private static readonly Action<Exception> PreserveStackTrace;
+
+        static ExceptionHelper()
+        {
+            var method = typeof(Exception).GetMethod("InternalPreserveStackTrace", BindingFlags.Instance | BindingFlags.NonPublic);
+
+            if (method != null)
+            {
+                try
+                {
+                    PreserveStackTrace = (Action<Exception>)Delegate.CreateDelegate(typeof(Action<Exception>), method);
+                    return;
+                }
+                catch (InvalidOperationException) { }
+            }
+            PreserveStackTrace = _ => { };
+        }
+#endif
+
+        /// <summary>
+        /// Rethrows an exception, preserving its stack trace
+        /// </summary>
+        /// <param name="exception">The exception to rethrow</param>
+        public static void Rethrow(Exception exception)
```

Comment:
For many years the policy of NUnit was to make lots of things public in order to be able to test them. This dates back to when we supported .NET versions where it wasn't possible to expose internals to your tests. the habit continues. Although the class is public, it's not part of any API, it's internal. People who use internal classes take their chances. Changing it would not be breaking but may be inconvenient for some of our tests.